File: app/src/main/java/hu/andras/daggersample/ui/feature2/list/Feature2ListPresenter.java
package hu.andras.daggersample.ui.feature2.list;

import android.os.Handler;

import hu.andras.daggersample.interactor.Feature2Interactor;
import hu.andras.daggersample.ui.common.Navigator;

/**
 * Created by Andras Nemeth on 2017. 06. 04..
 */

public class Feature2ListPresenter {

    private Feature2Interactor interactor;
    private Navigator navigator;

    public Feature2ListPresenter(Feature2Interactor interactor, Navigator navigator) {
        this.interactor = interactor;
        this.navigator = navigator;
    }

    void setView(Feature2ListActivity view) {
        view.display(interactor.getFeature2Stuff());
        new Handler().postDelayed(new Runnable() {
            @Override
            public void run() {
                navigator.goToFeature2DetailScreen();
            }
        }, 2000);
    }
}

File: app/src/main/java/hu/andras/daggersample/ui/feature2/list/Feature2ListPresenter.kt
package hu.andras.daggersample.ui.feature2.list

import hu.andras.daggersample.interactor.Feature2Interactor
import hu.andras.daggersample.ui.common.Navigator

/**
 * Created by Andras Nemeth on 2017. 06. 04..
 */
class Feature2ListPresenter(private val interactor: Feature2Interactor, private val navigator: Navigator) {
    fun setView(view: Feature2ListActivity) {
        view.display(interactor.feature2Stuff)
    }

    fun goToDetail() {
        navigator.goToFeature2DetailScreen()
    }
}
